pkgs/development/tools/build-managers/cmake/default.nix

```
@@ -51,7 +52,8 @@ stdenv.mkDerivation rec {
     ++ lib.optionals useSharedLibraries [ bzip2 curlMinimal expat libarchive xz zlib libuv rhash ]
     ++ lib.optional useOpenSSL openssl
     ++ lib.optional useNcurses ncurses
-    ++ lib.optional withQt5 qtbase;
+    ++ lib.optional withQt5 qtbase
+    ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.SystemConfiguration;
```

```suggestion
    ++ lib.optional stdenv.isDarwin SystemConfiguration;
```
We don't use darwin as an input for frameworks.

---

That was my first attempt, but
```
 $ nix-build -A pkgs.cmake                                unzip
error: anonymous function at /home/tobim/projects/nixpkgs/pkgs/development/tools/build-managers/cmake/default.nix:1:1 called without required argument 'SystemConfiguration'

       at /home/tobim/projects/nixpkgs/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|
(use '--show-trace' to show detailed location information)
```
I suspect `libsForQt5.callPackage` as the culprit, but didn't really investigate.

---

You need to add something like to https://github.com/NixOS/nixpkgs/blob/9811793e37e41b267a2c2bc2f8315e8d4910ecb3/pkgs/build-support/rust/default-crate-overrides.nix#L7 to all-packages.

---

That argument will probably be missing in the stdenv bootstrap phases. That might warrant
```nix
, SystemConfiguration ? null
```

---

This need a comment explaining the situation or it will be removed with the next occasion.

---

> That argument will probably be missing in the `stdenv` bootstrap phases. That might warrant

I figured as much, but didn't get around to fix it until now.

> This need a comment explaining the situation or it will be removed with the next occasion.

I opted to pass `null` explicitly in all-packages.nix and added a comment there.